File: maze.py
class Maze:
    def __init__(self,map):
        self.map = map
    
    def display(self):
        for i in range(len(self.map)):
            print(self.map[i])

    def findWinningSpot(self):
        for i in range(len(self.map)):
            for j in range(len(self.map[i])):
                if self.map[i][j] == 'E':
                    return [i, j]
    
    def findStartingPoint(self):
        for i in range(len(self.map)):
            for j in range(len(self.map[i])):
                if self.map[i][j] == 'P':
                    return [i, j]  

File: app/js/maze.js
export class Maze {
    constructor(map) {
        this.map = map;
    }

    display() {
        for (let i = 0; i < this.map.length; i++) {
            console.log(this.map[i]);
        }
    }

    findWinningSpot() {
        for (let i = 0; i < this.map.length; i++) {
            for (let j = 0; j < this.map[i].length; j++) {
                if (this.map[i][j] == 'E') {
                    return [i,j];
                }
            }
        }
    }

    findStartingPoint() {
        for (let i = 0; i < this.map.length; i++) {
            for (let j = 0; j < this.map[i].length; j++) {
                if (this.map[i][j] == 'P') {
                    return [i,j]
                }
            }
        }
    }
}